include/libcyphal/transport/can/media.hpp

```diff
@@ -68,20 +68,22 @@ class IMedia
     ///
     /// @return Returns `nullopt` on success; otherwise some `MediaError` in case of a low-level error.
     ///
-    CETL_NODISCARD virtual cetl::optional<MediaError> setFilters(const Filters filters) noexcept = 0;
+    virtual cetl::optional<MediaError> setFilters(const Filters filters) noexcept = 0;
 
     /// @brief Schedules the frame for transmission asynchronously and return immediately.
     ///
     /// @return Returns `true` if accepted or already timed out; `false` to try again later.
     ///
-    CETL_NODISCARD virtual Expected<bool, MediaError> push(const TimePoint                    deadline,
-                                                           const CanId                        can_id,
-                                                           const cetl::span<const cetl::byte> payload) noexcept = 0;
+    virtual Expected<bool, MediaError> push(const TimePoint                    deadline,
+                                            const CanId                        can_id,
+                                            const cetl::span<const cetl::byte> payload) noexcept = 0;
 
     /// @brief Takes the next payload fragment (aka CAN frame) from the reception queue unless it's empty.
     ///
     /// @param payload_buffer The payload of the frame will be written into the mutable `payload_buffer` (aka span).
     /// @return Description of a received fragment if available; otherwise an empty optional is returned immediately.
+    ///         `nodiscard` is used to prevent ignoring the return value, which contains not only possible media error,
+    ///         but also important metadata (like `payload_size` field for further parsing of the result payload).
```

Comment:
IMO this is the only public method which has little sense with discarding its result. All other methods could be used without checking results (f.e. if you don't care about possible errors and etc.) - anyway, no leaks or dangerous behavior will happen.